src/main/kotlin/io/github/mojira/arisa/modules/LanguageModule.kt

```diff
@@ -0,0 +1,58 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import arrow.core.left
+import arrow.core.right
+import com.github.pemistahl.lingua.api.Language
+import com.github.pemistahl.lingua.api.LanguageDetectorBuilder
+
+class LanguageModule(
+    private val allowedLanguages: List<Language> = listOf(
+        Language.ENGLISH,
+        Language.UNKNOWN
+    )
+) : Module<LanguageModule.Request> {
+
+    companion object {
+        private val detector = LanguageDetectorBuilder.fromLanguages(
+            Language.CHINESE,
+            Language.ENGLISH,
+            Language.FRENCH,
+            Language.GERMAN,
+            Language.JAPANESE,
+            Language.KOREAN,
+            Language.POLISH,
+            Language.PORTUGUESE,
+            Language.RUSSIAN,
+            Language.SPANISH
+        ).build()
+    }
+
+    data class Request(
+        val summary: String?,
+        val description: String?,
+        val resolveAsInvalid: () -> Either<Throwable, Unit>,
+        val addLanguageComment: (detectedLanguage: Language) -> Either<Throwable, Unit>
+    )
+
+    override fun invoke(request: Request): Either<ModuleError, ModuleResponse> = with(request) {
+        Either.fx {
+            val detectedLanguage = getDetectedLanguage(request.summary, request.description)
+
+            assertLanguageIsNotAllowed(allowedLanguages, detectedLanguage).bind()
+
+            addLanguageComment(detectedLanguage).toFailedModuleEither().bind()
+            resolveAsInvalid().toFailedModuleEither().bind()
+        }
+    }
+
+    private fun getDetectedLanguage(summary: String?, description: String?): Language {
+        return detector.detectLanguageOf("${summary ?: ""} ${description ?: ""}")
+    }
```

Comment:
I think it would make sense to not have this function in the actual module.
For jira lib stuff, we have `Jira.kt`, which makes it quite convenient for the module, especially when testing. Some of your current tests try to test the accuracy of the module instead, which in my opinion doesn't belong there.

I'd rather have tests that we can actually control, and test the accuracy of the lib on the tracker with hidden comments instead.